### PR TITLE
Increased the number of state variable accepted by Recurrence

### PR DESCRIPTION
**Pull request by delzac**
Increase the limit of the number of state variables accepted by Recurrence from 3 to 10.

### CODE REVIEW

File: bindings/python/cntk/layers/sequence.py [hunk 1]
```diff
@@ -299,8 +299,24 @@ def recurrence_from_2(h, c, x):
...
+        return _recurrence_from_n(h, c, a, b, d, x)
+    def recurrence_from_6(h, c, a, b, d, e, x):
+        return _recurrence_from_n(h, c, a, b, d, e, x)
+    def recurrence_from_7(h, c, a, b, d, e, f, x):
+        return _recurrence_from_n(h, c, a, b, d, e, f, x)
+    def recurrence_from_8(h, c, a, b, d, e, f, g, x):
+        return _recurrence_from_n(h, c, a, b, d, e, f, g, x)
+    def recurrence_from_9(h, c, a, b, d, e, f, g, i, x):
+        return _recurrence_from_n(h, c, a, b, d, e, f, g, i, x)
+    def recurrence_from_10(h, c, a, b, d, e, f, g, j, x):
```
Comment (KeDengMS) [inline bindings/python/cntk/layers/sequence.py]: missing i?

Comment (delzac) [reply bindings/python/cntk/layers/sequence.py]: Hi, sorry for the gross error. Amended and this good to go!

**Comment by KeDengMS** [comment]
Seems there's a testcase needs to be updated:
```
14:41:25 CNTKv2Python/ModuleTests: ___________________________ test_recurrence_step_fun ___________________________
14:41:25 CNTKv2Python/ModuleTests: 
14:41:25 CNTKv2Python/ModuleTests:     def test_recurrence_step_fun():
14:41:25 CNTKv2Python/ModuleTests:         import cntk as C
14:41:25 CNTKv2Python/ModuleTests:         def step_f(prev1, x):
14:41:25 CNTKv2Python/ModuleTests:             return prev1 * x
14:41:25 CNTKv2Python/ModuleTests:         rec = Recurrence(step_f)
14:41:25 CNTKv2Python/ModuleTests:     
14:41:25 CNTKv2Python/ModuleTests:         def step_f(prev1, prev2, x):
14:41:25 CNTKv2Python/ModuleTests:             return prev1 * prev2 * x, prev1 * x
14:41:25 CNTKv2Python/ModuleTests:         rec = Recurrence(step_f)
14:41:25 CNTKv2Python/ModuleTests:     
14:41:25 CNTKv2Python/ModuleTests:         def step_f(prev1, prev2, prev3, x):
14:41:25 CNTKv2Python/ModuleTests:             return prev1 * prev2 * prev3 * x, prev1 * x, prev2 * x
14:41:25 CNTKv2Python/ModuleTests:         rec = Recurrence(step_f)
14:41:25 CNTKv2Python/ModuleTests:     
14:41:25 CNTKv2Python/ModuleTests:         with pytest.raises(ValueError):
14:41:25 CNTKv2Python/ModuleTests:             def step_f(prev1, prev2, prev3, prev4, x):
14:41:25 CNTKv2Python/ModuleTests:                 return prev1 * prev2 * prev3 * x, prev1 * x, prev2 * x, prev4 * x
14:41:25 CNTKv2Python/ModuleTests: >           rec = Recurrence(step_f)
14:41:25 CNTKv2Python/ModuleTests: E           Failed: DID NOT RAISE <type 'exceptions.ValueError'>
14:41:25 CNTKv2Python/ModuleTests: 
14:41:25 CNTKv2Python/ModuleTests: /home/ubuntu/anaconda-envs/cntk-py27/lib/python2.7/site-packages/cntk/layers/tests/layers_test.py:182: Failed
14:41:25 CNTKv2Python/ModuleTests: ============= 1 failed, 1914 passed, 57 skipped in 473.84 seconds ==============
14:41:25 Test finished CNTKv2Python/ModuleTests (release cpu py27) - [FAILED] 479.88 sec
```

**Comment by delzac** [comment]
@KeDengMS Thanks for highlighting, sorry for the lapse! Done!

**Pull request merged**